docker/controller/Dockerfile

```diff
@@ -1,8 +1,32 @@
-FROM centos
+# Start from the latest golang base image
+FROM golang:latest as builder
```

Comment:
Please pin the image version (suggest `1.13`). Pinning is almost always a good thing.

In addition to this, if you're using Alpine runtime image you also need to build with an Alpine-based image. It has different core system libraries that Go will link against. Otherwise it may result in undefined behaviour.